Userland/Libraries/LibWeb/HTML/BrowsingContext.cpp

```diff
@@ -693,76 +695,101 @@ BrowsingContext* BrowsingContext::choose_a_browsing_context(StringView name, boo
     // recently opened, most recently focused, or more closely related.
-    if (!name.equals_ignoring_case("_blank"sv)) {
+    else if (!name.equals_ignoring_case("_blank"sv)) {
+        dbgln("FIXME: Find matching browser context for name {}", name);
         chosen = this;
     } else {
         // 8. Otherwise, a new browsing context is being requested, and what
         // happens depends on the user agent's configuration and abilities — it
         // is determined by the rules given for the first applicable option from
         // the following list:
-        dbgln("FIXME: Create a new browsing context!");
 
         // --> If current's active window does not have transient activation and
         //     the user agent has been configured to not show popups (i.e., the
         //     user agent has a "popup blocker" enabled)
-        //
-        //     The user agent may inform the user that a popup has been blocked.
+        VERIFY(m_page);
+        if (!active_window()->has_transient_activation() && m_page->should_block_pop_ups()) {
+            // FIXME: The user agent may inform the user that a popup has been blocked.
+            dbgln("Pop-up blocked!");
+        }
 
         // --> If sandboxingFlagSet has the sandboxed auxiliary navigation
         //     browsing context flag set
-        //
-        //     The user agent may report to a developer console that a popup has
-        //     been blocked.
+        else if (sandboxing_flag_set.flags & SandboxingFlagSet::SandboxedAuxiliaryNavigation) {
+            // FIXME: The user agent may report to a developer console that a popup has been blocked.
+            dbgln("Pop-up blocked!");
+        }
 
         // --> If the user agent has been configured such that in this instance
         //     it will create a new browsing context
-        //
-        //     1. Set windowType to "new and unrestricted".
-
-        //     2. If current's top-level browsing context's active document's
-        //     cross-origin opener policy's value is "same-origin" or
-        //     "same-origin-plus-COEP", then:
-
-        //         2.1. Let currentDocument be current's active document.
-
-        //         2.2. If currentDocument's origin is not same origin with
-        //         currentDocument's relevant settings object's top-level
-        //         origin, then set noopener to true, name to "_blank", and
-        //         windowType to "new with no opener".
-
-        //     3. If noopener is true, then set chosen to the result of creating
-        //     a new top-level browsing context.
+        else if (true) { // FIXME: When is this the case?
+            // 1. Set windowType to "new and unrestricted".
+            window_type = WindowType::NewAndUnrestricted;
+
+            // 2. If current's top-level browsing context's active document's
+            // cross-origin opener policy's value is "same-origin" or
+            // "same-origin-plus-COEP", then:
```

Comment:
Align wrapped lines with step number please

Reply:
Also, we currently go up to ~120 characters before wrapping, this seems overly aggressive (same below)